draft-ietf-quic-recovery.md

```diff
@@ -751,6 +751,11 @@ twice the maximum datagram size. This follows the analysis and recommendations
 in {{?RFC6928}}, increasing the byte limit to account for the smaller 8 byte
 overhead of UDP compared to the 20 byte overhead for TCP.
 
+If the maximum datagram size is decreased in order to complete the handshake,
+the initial congestion window SHOULD be recalculated with the new smaller size.
+This avoids overly large initial congestion windows.  The initial window is
+not recalculated at any other time.
```

Comment:
So if we are forced to reset to the initial window after going idle, we can't reset the congestion controller and calculate an initial using the current path MTU?  It seems to me like you might reasonably assume that an MTU is viable for longer than a congestion window.

Reply:
I agree with your second sentence, but I'm a bit confused about what you're asking.

The intent of the text to to say if you start with max datagram size X and drop down to X/2, because packets of size X aren't making it to the peer, then you should recalculate the initial window based on X/2.

Reply:
So I'm concerned about other places we use the initial window.  One is where you go idle and decide to reset the congestion window.  Another is where you move to a new path.  The former is interesting in this case because the MTU might carry across.  The latter, less so.

Reply:
Is your specific concern that you pick a max datagram size during the handshake, and then do PMTUD and find the path supports a larger datagram size, but you never increase the initial congestion window?

Reply:
I agree with @martinthomson. This can be made more general than just the handshake. Whenever there is a use of an IW, for bursts as an example, this recommendation applies. Is there a reason to make it specific to the handshake only?

Reply:
Not really, I was attempting to make the smallest change that fixed the issue.

Reply:
My comment related to the handshake because I saw that as most important. I agree though with Jana, a change of PMTU can require that the IW is updated, given that IW is also used after the handshake and might be cached for future connections. Could we change the final sentence in the PR to say something like: The initial window SHOULD also be recalculated when a change of the path MTU is detected.

Reply:
@gorryfair I updated the PR to be more general and the first sentence now says
"If the maximum datagram size changes during the connection, the initial
congestion window SHOULD be recalculated with the new size."